### PR TITLE
update: add missing exception, required_engine_version, required_plugin_version to -L json output

### PR DESCRIPTION
**Pull request by loresuso**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> /kind release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:
As per PR title

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update: add missing exception, required_engine_version, required_plugin_version to -L json output
```

### CODE REVIEW

**Review by jasondellaluce — COMMENTED** [review]
Thanks Lore! Two things missing before a LGTM:
1. The block at line 581 needs to be reworked to use `create_formatter(r.source, r.output)`, so that the engine creates the formatter with the wrong source. This is blocked by https://github.com/falcosecurity/libs/pull/1120, otherwise we'd be forced to do dynamic casting which would be gross.
2. When a rule/macro match all the event types, return all the types instead of an empty sequence (makes life sooo much easier on automatic checkers)

**Comment by FedeDP** [comment]
@loresuso can you add a release note?

File: cmake/modules/falcosecurity-libs.cmake [hunk 1]
```diff
@@ -27,8 +27,8 @@ else()
   # In case you want to test against another falcosecurity/libs version (or branch, or commit) just pass the variable -
   # ie., `cmake -DFALCOSECURITY_LIBS_VERSION=dev ..` 
   if(NOT FALCOSECURITY_LIBS_VERSION)
-    set(FALCOSECURITY_LIBS_VERSION "0.11.0-rc3")
-    set(FALCOSECURITY_LIBS_CHECKSUM "SHA256=92d78d64dbbdcbf08256be91e1b74f4d99caaea35f27324219fc6a5c195c99b8")
+    set(FALCOSECURITY_LIBS_VERSION "891862b0425a63ab48ca10399d6c5a1a2b8c4c95")
```
Comment (FedeDP) [inline cmake/modules/falcosecurity-libs.cmake:30]: Can we bump to 0.11.0-rc4 here?

**Review by FedeDP — APPROVED** [review]
/approve

File: userspace/engine/json_evt.h [hunk 2]
```diff
@@ -435,6 +435,10 @@ class json_event_formatter : public gen_event_formatter
 	bool tostring(gen_event *evt, std::string &output) override;
 	bool tostring_withformat(gen_event *evt, std::string &output, gen_event_formatter::output_format of) override;
 	bool get_field_values(gen_event *evt, std::map<std::string, std::string> &fields) override;
+	void get_field_names(std::vector<std::string> &fields) override 
+	{
+		throw falco_exception("json_event_formatter::get_field_names operation not supported");
```
[the remark this answers is not in the document]
Comment (loresuso) [reply userspace/engine/json_evt.h:441]: you are right, thanks Luca!

**Review by jasondellaluce — APPROVED** [review]
/approve

LGTM

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**